ultranest/popstepsampler.py

```diff
@@ -688,7 +688,333 @@ def __next__(
+                 
+                
+                # limite of the slice based on the unit cube boundaries
+                tleft_unitcube,tright_unitcube= unitcube_line_intersection(allu, v)
+                # Defining bound of the slice
+                tleft_worker,tright_worker = self.slice_limit(tleft_unitcube,tright_unitcube)
+                tleft,tright = self.slice_limit(tleft_unitcube,tright_unitcube)
+                # Index of the workers working concurrently
+                worker_running=np.arange(0,self.popsize,1,dtype=int)
+                # Status indicating if a points has already find its next position
+                status=np.zeros(self.popsize,dtype=int) # one for success, zero for running
+                
+
+                # Loop until each points has found its next position or we reached 100 iterations
+                
+                for it in range(self.max_it):
+                
+                    # Sampling points on the slices
+                    slice_position = np.random.uniform(size=(self.popsize,))
+                    
+                    t=tleft_worker+(tright_worker-tleft_worker)*slice_position
+                    
+                    
+                    
+                    points=allu[worker_running,:]
+                    v_worker=v[worker_running,:]
+                    proposed_u=points+t.reshape((-1,1))*v_worker
+                
+                    proposed_p = transform(proposed_u)
+                    proposed_L = loglike(proposed_p)
+                    nc+=self.popsize
+                    # Updating the pool of points based on the newly sampled points
+                    tleft,tright,worker_running,status,allu,allL,allp,n_discarded_it=update_vectorised_slice_sampler(\
+                    t,tleft,tright,proposed_L,proposed_u,proposed_p,worker_running,status,Lmin,self.shrink_factor,\
+                    allu,allL,allp,self.popsize)
+                    n_discarded+=n_discarded_it
+                    # Update of the limits of the slices
+                    tleft_worker=tleft[worker_running]
+                    tright_worker=tright[worker_running]
+                    if not np.any(status==0):
+                        break
+                # Record of the final interval on theta for scale adaptation
+                interval_final+=np.median(tright-tleft)
+
+
+            
+            interval_final=interval_final/self.nsteps
+            
+            
+            self.discarded+=n_discarded
+            self.ncalls+=nc
+            
+            assert np.array([p!=np.zeros(ndim) for p in allp]).all(), 'some walkers never moved! Double nsteps of PopulationSimpleSliceSampler.'
+            far_enough, (move_distance, reference_distance) = diagnose_move_distances(region, us[ilive,:], allu)
+            self.prepared_samples = list(zip(allu, allp, allL))
+
+            self.logstat.append([
+                1., # we always find a point or we will break at the assert before
```

Comment:
orient yourself based on here:
https://github.com/JohannesBuchner/UltraNest/blob/master/ultranest/popstepsampler.py#L558
* acceptance rate: number of accepted points divided by likelihood evaluations